File: ios/src/main/java/y2k/joyreactor/AddTagViewController.java
package y2k.joyreactor;

import org.robovm.apple.uikit.*;
import org.robovm.objc.annotation.CustomClass;
import org.robovm.objc.annotation.IBOutlet;
import y2k.joyreactor.presenters.AddTagPresenter;

/**
 * Created by y2k on 08/10/15.
 */
@CustomClass("AddTagViewController")
public class AddTagViewController extends UIViewController implements AddTagPresenter.View {

    UITextField tagNameView;
    UIActivityIndicatorView activityView;

    @Override
    public void viewDidLoad() {
        super.viewDidLoad();
        AddTagPresenter presenter = new AddTagPresenter(this);

        getNavigationItem().getRightBarButtonItem()
                .setOnClickListener(sender -> presenter.add(tagNameView.getText()));

        tagNameView.setDelegate(new UITextFieldDelegateAdapter() {

            @Override
            public boolean shouldReturn(UITextField textField) {
                presenter.add(tagNameView.getText());
                return false;
            }
        });

        tagNameView.becomeFirstResponder();
        activityView.stopAnimating();
    }

    @Override
    public void setIsBusy(boolean isBusy) {
        getNavigationItem().getRightBarButtonItem().setEnabled(!isBusy);
        getNavigationItem().setHidesBackButton(isBusy, true);
        if (isBusy) activityView.startAnimating();
        else activityView.stopAnimating();
    }

    @Override
    public UIStatusBarStyle getPreferredStatusBarStyle() {
        return UIStatusBarStyle.LightContent;
    }

    @IBOutlet
    void setTagNameView(UITextField tagNameView) {
        this.tagNameView = tagNameView;
    }

    @IBOutlet
    void setActivityView(UIActivityIndicatorView activityView) {
        this.activityView = activityView;
    }
}

File: ios/src/main/kotlin/y2k/joyreactor/AddTagViewController.kt
package y2k.joyreactor

import org.robovm.apple.uikit.UIActivityIndicatorView
import org.robovm.apple.uikit.UIStatusBarStyle
import org.robovm.apple.uikit.UITextField
import org.robovm.apple.uikit.UIViewController
import org.robovm.objc.annotation.CustomClass
import org.robovm.objc.annotation.IBOutlet
import y2k.joyreactor.common.ServiceLocator
import y2k.joyreactor.common.bindingBuilder
import y2k.joyreactor.viewmodel.AddTagViewModel

/**
 * Created by y2k on 08/10/15.
 */
@CustomClass("AddTagViewController")
class AddTagViewController : UIViewController() {

    @IBOutlet lateinit var tagNameView: UITextField
    @IBOutlet lateinit var activityView: UIActivityIndicatorView

    override fun viewDidLoad() {
        super.viewDidLoad()

        val vm = ServiceLocator.resolve<AddTagViewModel>()
        bindingBuilder(this) {
            action(vm.isBusy) {
                navigationItem.rightBarButtonItem.isEnabled = !it
                navigationItem.setHidesBackButton(it, true)
                if (it) activityView.startAnimating()
                else activityView.stopAnimating()
            }
            navigationItem {
                rightCommand { vm.add() }
            }
            textField(tagNameView, vm.tag)
        }
        tagNameView.becomeFirstResponder()
    }

    override fun getPreferredStatusBarStyle(): UIStatusBarStyle {
        return UIStatusBarStyle.LightContent
    }
}